rust/src/parallel_map.rs

```diff
@@ -125,16 +126,15 @@ where
         let (par_map, thread) = ThreadCommunication::<I::Item, T>::new_pair();
         communication.push(par_map);
         let handle = std::thread::spawn(move || loop {
-            match thread.receive.recv() {
-                Ok(optional_task) => match optional_task {
+            while let Ok(optional_task) = thread.receive.recv() {
```

Comment:
Actually, since you match `optional_task` directly below, you can also do
```suggestion
            while let Ok(Some(task)) = thread.receive.recv() {
```
and save a few more lines